### PR TITLE
Implement course sections

### PR DESCRIPTION
**Pull request by thederekkaplan**
Adds an optional argument `section-number` to `/course join` that allows user to join specific section of course.

### Todo:
- [x] Add documentation
- [x] Allow switching section without leaving/rejoining course
- [x] Add command to list classmates
- [x] Add optional argument `section-number` to `/course roster` to get only members of a specific section

### CODE REVIEW

File: src/commands/course/subcommands/join.ts [hunk 1]
```diff
@@ -14,10 +14,19 @@ const join = new SubCommand({
       required: true,
       type: Discord.CommandOptionType.ROLE,
     }),
+    new CommandOption({
+      name: 'section-number',
```
Comment (jaysella) [inline src/commands/course/subcommands/join.ts]: Should these just be `section`?

File: src/commands/course/subcommands/switchSection.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      required: true,
+      type: Discord.CommandOptionType.INTEGER,
+    }),
+  ],
+  handler: async ctx => {
+    const guildId = ctx.mustGetGuildId();
+    const roleId = ctx.getArgument<string>('role') as string;
+    const sectionNum = parseInt(
+      ctx.getArgument<string>('section-number') as string,
+    );
```
Comment (jaysella) [inline src/commands/course/subcommands/switchSection.ts]: 1. Should there be a regex validation here like used for course numbers? For example, returning an error message if the section number provided is not a number or if too few/many digits.
2. Also, is there any standardization with leading zeros? Sections `1` and `01` should be saved as the same.

**Review by jaysella — CHANGES_REQUESTED** [review]
Should there be a `get-section` command?

It might also be worth updating the course commands to include the new `Permissions` column.

File: src/commands/course/subcommands/switchSection.ts (hunk 2 above)
Comment (thederekkaplan) [reply src/commands/course/subcommands/switchSection.ts]: It's `Discord.CommandOptionType.INTEGER`, so it's guaranteed to be an integer. And parseInt ignores leading zeros.

**Comment by thederekkaplan** [comment]
@j-651 What would a `get-section` command do?

**Pull request merged**